### PR TITLE
Added REST API link in file JAVA doc

### PR DESCRIPTION
**Pull request by sima-zhu**
Minor fix for signature name
Minor fix for javadoc indentation
Added REST API link for file

### CODE REVIEW

**Comment by sima-zhu** [comment]
@alzimmermsft Could you take a look at again and leave comments for what you have done for the queue PR? I would like to keep consistent with Queue. Thanks!

**Comment by rickle-msft** [comment]
>     public Flux<HandleItem> getHandles(Integer maxResult, boolean recursive) {

This name doesn't seem right. I thought we didn't use "get" for apis that make service calls. Unless there's some reason why we deviated here, can we open an item to change this name?

---
Refers to: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java:298 in ed97215. [](commit_id = ed97215c61ebd50428e88e34f68a7fa12f65b44c, deletion_comment = False)

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java [hunk 1]
```diff
@@ -91,6 +91,9 @@ public DirectoryClient getSubDirectoryClient(String subDirectoryName) {
      *
      * {@codesnippet com.azure.storage.file.directoryClient.createDirectory}
      *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: >/list-directories-and-files" [](start = 73, length = 28)

Incorrect link

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java [hunk 2]
```diff
@@ -107,6 +110,9 @@ public DirectoryClient getSubDirectoryClient(String subDirectoryName) {
      *
      * {@codesnippet com.azure.storage.file.directoryClient.create#map}
      *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: >list-directories-and-files [](start = 74, length = 26)

Incorrect link

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java [hunk 3]
```diff
@@ -248,6 +275,9 @@ public VoidResponse delete() {
      *
      * {@codesnippet com.azure.storage.file.directoryClient.createSubDirectory#string}
      *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: >list-directories-and-files [](start = 74, length = 26)

Incorrect link

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java [hunk 4]
```diff
@@ -265,6 +295,9 @@ public VoidResponse delete() {
      *
      * {@codesnippet com.azure.storage.file.directoryClient.createSubDirectory#string-map}
      *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: >list-directories-and-files [](start = 74, length = 26)

incorrect link

**Comment by rickle-msft** [comment]
>      * {@codesnippet com.azure.storage.file.fileClient.downloadWithProperties}

Why no link here? It looks like there are a couple methods that don't have docs

---
Refers to: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileClient.java:189 in ed97215. [](commit_id = ed97215c61ebd50428e88e34f68a7fa12f65b44c, deletion_comment = False)

**Comment by sima-zhu** [comment]
> > ```
> >  * {@codesnippet com.azure.storage.file.fileClient.downloadWithProperties}
> > ```
> 
> Why no link here? It looks like there are a couple methods that don't have docs
> 
> Refers to: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileClient.java:189 in [ed97215](https://github.com/Azure/azure-sdk-for-java/commit/ed97215c61ebd50428e88e34f68a7fa12f65b44c). [](commit_id = [ed97215](https://github.com/Azure/azure-sdk-for-java/commit/ed97215c61ebd50428e88e34f68a7fa12f65b44c), deletion_comment = False)

I do not find the appropriate link for this API, same applies to others which leave blank.

**Comment by rickle-msft** [comment]
Should just be the download api

---
In reply to: [517047843](https://github.com/Azure/azure-sdk-for-java/pull/4568#issuecomment-517047843) [](ancestors = 517047843)

**Comment by rickle-msft** [comment]
The description says you bumped the version number, but I only see edits to Java files?

**Comment by sima-zhu** [comment]
> The description says you bumped the version number, but I only see edits to Java files?

That has been done by other PR. Will remove the description. @rickle-msft

**Comment by sima-zhu** [comment]
> Should just be the download api
> 
> In reply to: [517047843](https://github.com/Azure/azure-sdk-for-java/pull/4568#issuecomment-517047843) [](ancestors = 517047843)

I only saw the getFile which is corresponding to downloadToFile. It does not provide properties for download.

**Comment by sima-zhu** [comment]
> > ```
> > public Flux<HandleItem> getHandles(Integer maxResult, boolean recursive) {
> > ```
> 
> This name doesn't seem right. I thought we didn't use "get" for apis that make service calls. Unless there's some reason why we deviated here, can we open an item to change this name?
> 
> Refers to: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java:298 in [ed97215](https://github.com/Azure/azure-sdk-for-java/commit/ed97215c61ebd50428e88e34f68a7fa12f65b44c). [](commit_id = [ed97215](https://github.com/Azure/azure-sdk-for-java/commit/ed97215c61ebd50428e88e34f68a7fa12f65b44c), deletion_comment = False)

@rickle-msft I had the method name from design doc. I do not have the background of why it names like this. I can create an issue for this. https://github.com/Azure/azure-sdk-for-java/issues/4754

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: Done with the checking for the entire file.

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java (hunk 2 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: Done.

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java (hunk 3 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: Done.

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java (hunk 4 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java]: Done.

**Pull request merged**